File: src/main/java/com/lessmarkup/interfaces/system/UserCache.java
package com.lessmarkup.interfaces.system;

import com.lessmarkup.interfaces.cache.CacheHandler;
import com.lessmarkup.interfaces.structure.CachedNodeInformation;
import com.lessmarkup.interfaces.structure.NodeAccessType;
import com.lessmarkup.interfaces.structure.Tuple;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.OptionalLong;

public interface UserCache extends CacheHandler {
    String getName();
    boolean isRemoved();
    boolean isAdministrator();
    boolean isApproved();
    boolean isEmailConfirmed();
    List<Long> getGroups();
    String getEmail();
    String getTitle();
    boolean isBlocked();
    OffsetDateTime getUnblockTime();
    String getProperties();
    OptionalLong getAvatarImageId();
    OptionalLong getUserImageId();
    List<Tuple<CachedNodeInformation, NodeAccessType>> getNodes();
}

File: src/main/scala/com/lessmarkup/interfaces/system/UserCache.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.system

import java.time.OffsetDateTime

import com.lessmarkup.interfaces.annotations.NodeAccessType
import com.lessmarkup.interfaces.cache.CacheHandler
import com.lessmarkup.interfaces.structure.CachedNodeInformation

trait UserCache extends CacheHandler {
  def getName: String

  def isRemoved: Boolean

  def isAdministrator: Boolean

  def isApproved: Boolean

  def isEmailConfirmed: Boolean

  def getGroups: Seq[Long]

  def getEmail: String

  def getTitle: String

  def isBlocked: Boolean

  def getUnblockTime: Option[OffsetDateTime]

  def getProperties: String

  def getAvatarImageId: Option[Long]

  def getUserImageId: Option[Long]

  def getNodes: Seq[(CachedNodeInformation, NodeAccessType)]
}